src/isl/isl_type.rs

```diff
@@ -1,61 +1,151 @@
 use crate::isl::isl_constraint::IslConstraint;
 use crate::isl::isl_type_reference::IslTypeRef;
-use crate::result::{
-    invalid_schema_error, invalid_schema_error_raw, IonSchemaError, IonSchemaResult,
-};
-use ion_rs::value::owned::OwnedElement;
+use crate::result::{invalid_schema_error, invalid_schema_error_raw, IonSchemaResult};
+use ion_rs::value::owned::{text_token, OwnedElement, OwnedSymbolToken};
 use ion_rs::value::{Element, Sequence, Struct, SymbolToken};
 use ion_rs::IonType;
-use std::convert::TryFrom;
 
-/// Represents a public facing schema type [IslType] which can be converted to a solid TypeRef using TypeStore
+/// Represents the public facing schema [IslType] which converts given ion content in the schema file
+/// into an internal representation of ISL types(not-yet-resolved types).
+#[derive(Debug, Clone, PartialEq)]
+pub enum IslType {
+    NamedIslType(NamedIslType),
+    AnonymousIslType(AnonymousIslType),
+}
+
+/// Represents a public facing schema type [AnonymousIslType] which can be converted to a solid [AnonymousTypeDefinition] using TypeStore
+/// anonymous ISL type grammar: `{ <CONSTRAINT>... }`
 #[derive(Debug, Clone)]
-pub struct IslType {
-    name: Option<String>,
+pub struct AnonymousIslType {
     constraints: Vec<IslConstraint>,
 }
 
-impl IslType {
-    pub fn new(name: Option<String>, constraints: Vec<IslConstraint>) -> Self {
-        Self { name, constraints }
-    }
-
-    pub fn name(&self) -> &Option<String> {
-        &self.name
+impl AnonymousIslType {
+    pub fn new(constraints: Vec<IslConstraint>) -> Self {
+        Self { constraints }
     }
 
     pub fn constraints(&self) -> &[IslConstraint] {
         &self.constraints
     }
+
+    /// Parse constraints inside an [OwnedElement] to an [AnonymousIslType]
+    pub fn parse_from_owned_element(ion: &OwnedElement) -> IonSchemaResult<Self> {
+        let mut constraints = vec![];
+
+        let ion_struct = try_to!(ion.as_struct());
+
+        // parses all the constraints inside a Type
+        for (field_name, value) in ion_struct.iter() {
+            let constraint_name = match field_name.text() {
+                Some("name") => continue, // if the field_name is "name" then it's the type name not a constraint
```

Comment:
ISL allows for open content, which means that you can stick a "name" field anywhere, but it will be ignored if it's not in a top-level type. Skipping it silently is what the spec requires.